### ISSUE TITLE
Possible TypeError: Cannot read property on server

### ISSUE THREAD

**Issue by hthetiot** [priologic/easyrtc#178]
@EricDavies I got that error after reloading a server for update.
I see a lot of "e.app[appName].connection[easyrtcid].room[roomName]" in the source that can cause "TypeError: Cannot read property " may be we should add a getter method for that to avoid TypeError in case of missing connection easyrtcid obj.

> app/node_modules/easyrtc/lib/easyrtc_public_obj.js:2146
>             if (e.app[appName].connection[easyrtcid].room[roomName]) {
>                                                     ^
> TypeError: Cannot read property 'room' of undefined
>     at Object.connectionObj.joinRoom (app/node_modules/easyrtc/lib/easyrtc_public_obj.js:2146:53)
>     at Query.<anonymous> (app/server/middlewares/easyrtc.js:473:43)
>     at app/node_modules/mongoose/node_modules/kareem/index.js:177:19
>     at app/node_modules/mongoose/node_modules/kareem/index.js:109:16
>     at doNTCallback0 (node.js:408:9)
>     at process._tickCallback (node.js:337:13)
> error: Forever detected script exited with code: 1

**Comment by EricDavies** [comment]
Hi Harold,

That's a server item, so I'm forwarding to Rod.

Eric.

**Comment by hthetiot** [comment]
I have not seeing that in a while.
Closing for now.

**Comment by hthetiot** [comment]
Just got the error did a push for fix

**Comment by chpasha** [comment]
the same error occasionally happens when leaving the room in the connectionObj.room = function(roomName, callback)  - it needs the same check as in pull request above

**Comment by hthetiot** [comment]
```
[2017-08-31 19:43:52]: warning [EasyRTC] Attempt to request non-existent connection key: 'WRJdhnyqbBOSfvQB'
[2017-08-31 19:43:52]: warning [EasyRTC] [8riW0o0DG6AkcGen] Could not send WebRTC signal to client [WRJdhnyqbBOSfvQB]. They may no longer be online.
[2017-08-31 19:44:32]: error   [Error] Possibly Unhandled Rejection at: Promise  { state: 'rejected',
  reason: [TypeError: Cannot read property 'room' of undefined] }  reason:  [TypeError: Cannot read property 'room' of undefined]

```

**Comment by hthetiot** [comment]
Fixed in 55b3030f902f10d40eca00867c539bfc0c141aab
